GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1157,12 +1157,35 @@ private void UpdatePlaceholders()
                     //  1. Ensures child folders are deleted before their parents
                     //  2. Ensures that folders that have been deleted by git (but are still in the projection) are found before their
                     //     parent folder is re-expanded (only applies on platforms where EnumerationExpandsDirectories is true)
+                    Dictionary<string, long> availableSizes = new Dictionary<string, long>();
                     foreach (PlaceholderListDatabase.PlaceholderData folderPlaceholder in placeholderFoldersListCopy.OrderByDescending(x => x.Path))
                     {
-                        // Remove folder placeholders before re-expansion to ensure that projection changes that convert a folder to a file work
-                        // properly
-                        if (!this.RemoveFolderPlaceholderIfEmpty(folderPlaceholder, addPlaceholderToUpdatedPlaceholders, folderPlaceholdersToKeep))
+                        bool keepFolder = true;
+                        if (!folderPlaceholdersToKeep.Contains(folderPlaceholder.Path))
                         {
+                            this.GetChildNameAndParentKey(folderPlaceholder.Path, out string childName, out string parentKey);
+                            FolderData folderData = this.GetProjectedFolderEntryData(CancellationToken.None, blobSizesConnection, availableSizes, childName, parentKey, out string _) as FolderData;
+

```

Comment:
It looks like `GetProjectedFolderEntryData` may attempt to download size(s) if the path we pass in is for a file.

Should/could we do one of the following instead:

- Use `IsPathProjected` here instead of `GetChildNameAndParentKey` + `GetProjectedFolderEntryData`
- Pass in `null` for `blobSizesConnection` to prevent `GetProjectedFolderEntryData` from downloading sizes

Reply:
Yeah looks like `IsPathProjected` would be best.